### ISSUE TITLE
Default 'Relevance' sort order missing from search block's `Sort on options`

### ISSUE THREAD

**Issue by JeffersonBledsoe** [plone/volto#4000]
**Describe the bug**
If a `sort_on` key is not specified, Plone will order a querystring search by 'relevance', which is often the desired behaviour. A search block enables site viewer sorting options and will default to the `relevance` sorting if one is not selected. Not only is it not possible to return to the `relevance` sorting once an option has been set, it isn't clear that Plone is even performing a `relevance` sort initially as it isn't mentioned anywhere. It would nice for site editors to be able to add a `Relevance` option to the search block's sort on options.

**To Reproduce**

1. Create a search block
2. Enable the `Show sorting` option in the `Controls` fieldset
3. Try to add an option to allow sorting by `relevance`

**Expected behavior**
An option for `relevance` is shown in the dropdown. This option would simply remove the `sort_on` key from the querystring, allowing Plone to return to it's default sort order.

**Comment by tiberiuichim** [comment]
@JeffersonBledsoe Would it work if the value for the "Relevance" option would be an empty string "" ?

**Comment by JeffersonBledsoe** [comment]
Exactly that! It's just not currently there as the options for sort on only display a list of available indexes. Just wanted some feedback on the UX before I put a PR together :)

**Comment by tiberiuichim** [comment]
I think having "Relevance" as an option would be just fine.

**Comment by djay** [comment]
It's not that simple because relevance only applies when there is a keyword search. If it's a listing with no search then empty just means random order.
I think it would be confusing to have the option of relevance if it's list a listing with no search.

So either the sort block is going to need a setting to enable the relevance option to be - [ ] shown or perhaps the words switch between "unsorted" and "relavance" depending if a keyword search has been entered?

**Comment by tiberiuichim** [comment]
@djay ok, so we actually need to hide the Relevance as an option if there's no search term (and automatically switch to the default sort option)

**Comment by davisagli** [comment]
@tiberiuichim If there's no search term and no sort specified, then the order of the results is nondeterministic (well it is probably determined by the internal structure of the index BTrees). Showing "unsorted" in that case is probably the best we can do.

**Comment by JeffersonBledsoe** [comment]
Just working on this now and realistically the behaviour is already there, it's just that the 'No selection' option in the sort on needs renaming and for it to be selectable. My first attempt at this (shown below) checked whether there was any text in the search box and changed the label between 'Relevance' and 'Unsorted', but this doesn't feel quite right that the relevance option seems to disappear. Does it maybe make sense to always show 'Relevance' if the search input is visible, but if it's hidden (i.e. filter only search block) then change the text to 'Unsorted'?

https://user-images.githubusercontent.com/30210785/206512157-39f68cf3-56cd-422e-8769-f28a5dc723c5.mov

**Comment by djay** [comment]
It's a little weird but not that weird and it's correct. Without a search the results are unsorted

**Comment by tiberiuichim** [comment]
I have the following proposal (I'm testing Jeff's PR):

- In the Query part of the search block, the "sort on" selected there will signify "default search on". It will be used when the search block is used with any entered SearchableText
- Whenever the user uses the search text input, we reset the search back to relevance

I'm attaching a screencast of using the https://www.eea.europa.eu/en/advanced-search (this is implemented using ElasticSearch with https://github.com/eea/volto-searchlib

[vokoscreenNG-2023-03-24_14-15-12.webm](https://user-images.githubusercontent.com/141568/227519107-a553bdac-e978-4355-8e1f-ccbf0f0c63a7.webm)

**Comment by djay** [comment]
@tiberiuichim I agree its nicer to use a default search when not using a keyword but I suspect it could be annoying to reset the sort on every keyword change. I think it might be better to switch to relevence only when you are going from blank query to a keyword one. e.g. I search for foo, sort on newest and then change the keyword to bah, it would be annoying if I then had to change the sort again to newest.

**Comment by tiberiuichim** [comment]
@djay we've used that behavior initially, in the EEA search engine, but we had a UX study and they came back that the focus group users forgot that they've had sorting options set and were surprised when the documents they searched for didn't match their expectations for relevancy. Of course, it could be that the EEA search engine is just busier with UI elements and the users didn't understand the sorting.

**Comment by tiberiuichim** [comment]
@djay Google is of the same opinion, that the relevance should be used as default whenever you change the search term.

[vokoscreenNG-2023-03-24_18-44-31.webm](https://user-images.githubusercontent.com/141568/227588607-7aaa744c-537d-4acb-8b02-c3e14796fe1d.webm)

**Comment by djay** [comment]
@tiberiuichim good point.

**Comment by djay** [comment]
Found another example that works the same way - https://www.gov.uk/search/all. Will switch to relevance if adding a keyword.

**Comment by djay** [comment]
@tiberiuichim I've updated the description to specify the following behavior and we will work on implementing this

> - if a keyword is added then switch the sort_on to show relevance.
> - if all keywords are removed then switch the sort_on to show the default sort (or maybe previous selection if thats easy to do)
>    - Relevance option should be removed from the dropdown. You can't pick it if there is no keywords.
> - if the user types a keyword, then changes the sort_on, then edits the keywords again... then we should still switch to sort_on back to relevance.
